next/app/coins/[userID]/route.ts

```diff
@@ -0,0 +1,39 @@
+import { headers } from "next/headers";
```

Comment:
[NEXT](https://nextjs.org/docs/pages/building-your-application/routing/api-routes) I think it is pretty standard for Next the use the file structure app/api/whatever instead of just the using the app router for APIs. I could be wrong on this but figured I would share my experience with this.

This structure allows for separation between pages and APIs.

Reply:
From what I have seen, that is usually for full stack applications when they share a frontend and backend in the same repo. Since this is just an api, we would have to configure it differently if we wanted the directory structure to not match the routing structure. For instance, you are testing for the endpoints `/test` or `/coins/{userID}`. Right now, I think it would have to route to `/api/test` of `/api/coins/{userID}` if we wanted that directory structure. 

View [here](https://nextjs.org/docs/app/building-your-application/routing/colocation) for more on routing.